osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/domain/services/ClientConnectionService.java

```diff
@@ -110,7 +111,11 @@ private ClientConnection createConnection(final RequestMetadata requestMetadata)
         final ClientConnection newDeviceConnection = this.iec60870Client.connect(connectionParameters, eventListener);
 
         try {
-            this.connectionCache.addConnection(deviceIdentification, newDeviceConnection);
+            if (device.hasGatewayDevice()) {
+                this.connectionCache.addConnection(connectionDeviceIdentification, newDeviceConnection);
+            } else {
+                this.connectionCache.addConnection(deviceIdentification, newDeviceConnection);
+            }
```

Comment:
I think this can be simplified to just:
this.connectionCache.addConnection(connectionDeviceIdentification, newDeviceConnection);

The call to this.getConnectionDeviceIdentification(device) already returns the identification of the device or its gateway, if it has one.